include/boost/compute/system.hpp

```diff
@@ -149,17 +160,21 @@ class system
     /// same context object being returned.
     static context default_context()
     {
-        static context default_context(default_device());
-
-        return default_context;
+        return init_default_context();
     }
 
     /// Returns the default command queue for the system.
-    static command_queue& default_queue()
+    ///
+    /// If user-provided command queue is given, the system-wide default context
+    /// and default device will be set up appropriately so that the default queue
+    /// matches the default context and device.
+    ///
+    /// The default queue is created once on the first time this function is
+    /// called. Calling this function multiple times will always result in the
+    /// same command queue object being returned.
```

Comment:
I would add information about potential exception if default context and.or default device don't match with device and context of `user_queue`.

Reply:
You can point to the test you created.

Reply:
Agreed. Will revise soon this week